### PR TITLE
Add option :all to :no_warn_undefined

### LINKED ISSUE #9676: Release steps defined in dependencies render warnings
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 1.10-dev (7f86ea3)
* Operating system: *

### Current behavior
`
Mix will render a warning about functions in the release steps that come from dependencies.

### Expected behavior

Do not render the warning. This was the behavior < 1.10

Here is an example project that demonstrates this: https://github.com/mobileoverlord/release_warn

### PR DESCRIPTION
**Pull request by wojtekmach**
Closes #9676

### CODE REVIEW

**Comment by ericmj** [comment]
Should we only support this as a compiler option and not on the `@compile` attribute?

**Comment by josevalim** [comment]
> Should we only support this as a compiler option and not on the @compile attribute?

Yes, I would say so for now. Because you should have control over what you are calling.

File: lib/elixir/lib/code.ex [hunk 1]
```diff
@@ -1049,6 +1049,16 @@ defmodule Code do
     :ok
   end
 
+  def put_compiler_option(:no_warn_undefined, value) do
+    if value != :all and not is_list(value) do
+      raise "compiler option :no_warn_undefined should be a list or an atom :all, " <>
```
Comment (whatyouhide) [inline lib/elixir/lib/code.ex]:
```suggestion
      raise "compiler option :no_warn_undefined should be a list or the atom :all, " <>
```

**Comment by ericmj** [comment]
Thank you Wojtek!

**Pull request merged**